src/common/auth/actions.js

```diff
@@ -10,17 +10,17 @@ const validateForm = (validate, fields) => validate(fields)
   .prop('password').required().simplePassword()
   .promise;
 
-const post = (fetch, endpoint, body) =>
-  fetch(`/api/v1/${endpoint}`, {
-    body: JSON.stringify(body),
-    headers: {'Accept': 'application/json', 'Content-Type': 'application/json'},
-    method: 'post'
-  })
-  .then(response => {
-    if (response.status === 200) return response.json();
-    throw response;
+const post = async (fetch, endpoint, body) => {
+  const response = await fetch(`/api/v1/${endpoint}`, {
+    method: 'POST',
+    headers: {'Content-Type': 'application/json'},
+    body: JSON.stringify(body)
   });
 
+  if (response.status === 200) return response.json();
+  throw response;
```

Comment:
Since in this case we've got the API in our hands, within the same project and server, IMHO it's ok to be handled this way.